### ISSUE TITLE
Inaccurate Solution - 787. Cheapest Flights Within K Stops

### ISSUE THREAD

**Issue by DavyC** [LeetCode-Feedback/LeetCode-Feedback#4171]
#### Your LeetCode username

fatezai

#### Category of the bug
- [ ] Question
- [X] Solution
- [ ] Language
- [ ] Missing Test Cases 

#### Description of the bug

In the first approach in the solution,
```
// Better cost?
if (dU + wUV < dV) {
    minHeap.offer(new int[]{nei, dU + wUV, stops + 1});
    distances[nei] = dU + wUV;
}
```
should be
```
// Better cost?
if (dU + wUV < dV) {
    minHeap.offer(new int[]{nei, dU + wUV, stops + 1});
    distances[nei] = dU + wUV;
    currentStops[nei] = stops;
}
```

#### Code you used for Submit/Run operation

```
class Solution {
    public int findCheapestPrice(int n, int[][] flights, int src, int dst, int k) {
        int[][] graph = new int[n][n];
        for (int[] flight : flights) {
            graph[flight[0]][flight[1]] = flight[2];
        }
        int[] accuPrices = new int[n];
        Arrays.fill(accuPrices, Integer.MAX_VALUE);
        accuPrices[src] = 0;
        int[] accuStops = new int[n];
        Arrays.fill(accuStops, Integer.MAX_VALUE);
        accuStops[src] = 0;
        PriorityQueue<int[]> queue = new PriorityQueue<int[]>((a, b) -> a[1] - b[1]);
        queue.add(new int[]{src, 0, 0});
        while (!queue.isEmpty()) {
            int[] data = queue.poll();
            if (data[0] == dst) {
                return data[1];
            }
            if (data[2] == k + 1) {
                continue;
            }
            int[] adjs = graph[data[0]];
            for (int i = 0; i < n; i++) {
                if (adjs[i] == 0) {
                    continue;
                }
                int nprice = data[1] + adjs[i];
                if (nprice < accuPrices[i]) {
                    queue.add(new int[] {i, nprice, data[2] + 1});
                    accuPrices[i] = nprice;
                    accuStops[i] = data[2];
                } else if (data[2] < accuStops[i]) {
                    queue.add(new int[] {i, nprice, data[2] + 1});
                    accuStops[i] = data[2];
                }
            }
        }
        return accuPrices[dst] == Integer.MAX_VALUE ? -1 : accuPrices[dst];
    }
}
```

#### Language used for code

JAVA

#### Expected behavior

Without stops being updated, when the new "`distance`" is the same as the recorded, we can't use the `stops` to compare if we didn't record it from the previous best `distance`.

Simply put, the provided solution didn't pass the following test case:
```
11
[[0,3,3],[3,4,3],[4,1,3],[0,5,1],[5,1,100],[0,6,2],[6,1,100],[0,7,1],[7,8,1],[8,9,1],[9,1,1],[1,10,1],[10,2,1],[1,2,100]]
0
2
4
```

**Comment by lc-kay** [comment]
Hi @DavyC 
Thank you for reaching out to us. I've relayed this issue to our team to investigate.

**Comment by LC-Sue** [comment]
Hi @DavyC 
Thank you for your feedback. The problem has already been updated. We appreciate your support!
